### ISSUE TITLE
aws - glue-catalog filter glue-security-config boolean error

### ISSUE THREAD

**Issue by soncaocc** [cloud-custodian/cloud-custodian#5697]
### Background
glue-catalog was recently introduced as a new resource type with some available encryption filters.
The filter `glue-security-config` for property `ReturnConnectionPasswordEncrypted` returns the following error when the policy is run:
`AttributeError: 'bool' object has no attribute 'startswith'`

### To Reproduce
Run the following policy on version 0.9.1
```yaml
policies:
  - name: glue-encryption
    resource: glue-catalog
    filters:
      - type: glue-security-config
        ReturnConnectionPasswordEncrypted: false
```

### Expected behavior
resources.json in output directory should contain json object of data catalog which has DataCatalogEncryptionSettings.ConnectionPasswordEncryption.ReturnConnectionPasswordEncrypted set to false.

**Background (please complete the following information):**
 - OS: OSX 10.14
 - Python Version: 3.7.6
 - Custodian Version: 0.9.1
 - Tool Version: [N/A]
 - Cloud Provider: aws
 - Policy: 
```yaml
policies:
  - name: glue-encryption
    resource: glue-catalog
    filters:
      - type: glue-security-config
        ReturnConnectionPasswordEncrypted: false
```
 - Traceback: 
```
Traceback (most recent call last):
  File "/Users/foobar/.local/share/virtualenvs/c7n-0.9.1-KpK8LEqQ/lib/python3.7/site-packages/c7n/policy.py", line 291, in run
    resources = self.policy.resource_manager.resources()
  File "/Users/foobar/.local/share/virtualenvs/c7n-0.9.1-KpK8LEqQ/lib/python3.7/site-packages/c7n/resources/glue.py", line 516, in resources
    return self.filter_resources(self._get_catalog_encryption_settings())
  File "/Users/foobar/.local/share/virtualenvs/c7n-0.9.1-KpK8LEqQ/lib/python3.7/site-packages/c7n/manager.py", line 109, in filter_resources
    resources = f.process(resources, event)
  File "/Users/foobar/.local/share/virtualenvs/c7n-0.9.1-KpK8LEqQ/lib/python3.7/site-packages/c7n/filters/multiattr.py", line 38, in process
    target = self.get_target(r)
  File "/Users/foobar/.local/share/virtualenvs/c7n-0.9.1-KpK8LEqQ/lib/python3.7/site-packages/c7n/resources/account.py", line 1577, in get_target
    if not self.data[kmskey].startswith('alias'):
AttributeError: 'bool' object has no attribute 'startswith'
2020-05-01 11:39:56,220: custodian.commands:ERROR Error while executing policy glue-encryption, continuing
Traceback (most recent call last):
  File "/Users/foobar/.local/share/virtualenvs/c7n-0.9.1-KpK8LEqQ/lib/python3.7/site-packages/c7n/commands.py", line 281, in run
    policy()
  File "/Users/foobar/.local/share/virtualenvs/c7n-0.9.1-KpK8LEqQ/lib/python3.7/site-packages/c7n/policy.py", line 1056, in __call__
    resources = PullMode(self).run()
  File "/Users/foobar/.local/share/virtualenvs/c7n-0.9.1-KpK8LEqQ/lib/python3.7/site-packages/c7n/policy.py", line 291, in run
    resources = self.policy.resource_manager.resources()
  File "/Users/foobar/.local/share/virtualenvs/c7n-0.9.1-KpK8LEqQ/lib/python3.7/site-packages/c7n/resources/glue.py", line 516, in resources
    return self.filter_resources(self._get_catalog_encryption_settings())
  File "/Users/foobar/.local/share/virtualenvs/c7n-0.9.1-KpK8LEqQ/lib/python3.7/site-packages/c7n/manager.py", line 109, in filter_resources
    resources = f.process(resources, event)
  File "/Users/foobar/.local/share/virtualenvs/c7n-0.9.1-KpK8LEqQ/lib/python3.7/site-packages/c7n/filters/multiattr.py", line 38, in process
    target = self.get_target(r)
  File "/Users/foobar/.local/share/virtualenvs/c7n-0.9.1-KpK8LEqQ/lib/python3.7/site-packages/c7n/resources/account.py", line 1577, in get_target
    if not self.data[kmskey].startswith('alias'):
AttributeError: 'bool' object has no attribute 'startswith'
```

 - `custodian version --debug` output
```
Custodian:   0.9.1
Python:      3.7.6 (default, Dec 30 2019, 19:38:28)
             [Clang 11.0.0 (clang-1100.0.33.16)]
Platform:    posix.uname_result(sysname='Darwin', nodename='xxxxx', release='18.7.0', version='Darwin Kernel Version 18.7.0: Mon Feb 10 21:08:45 PST 2020; root:xnu-4903.278.28~1/RELEASE_X86_64', machine='x86_64')
Using venv:  True
Docker: False
Installed:

argcomplete==1.11.1
attrs==19.3.0
boto3==1.12.47
botocore==1.15.47
docutils==0.15.2
importlib-metadata==1.6.0
jmespath==0.9.5
jsonschema==3.2.0
pyrsistent==0.16.0
python-dateutil==2.8.1
pyyaml==5.3.1
s3transfer==0.3.3
setuptools==46.1.3
six==1.14.0
tabulate==0.8.7
urllib3==1.25.9
zipp==3.1.0
```

**Additional context**

**Comment by PratMis** [comment]
Thanks for the bug report! I see what the problem is here. Will put out a Pr soon to address this!

**Comment by JohnHillegass** [comment]
Closed by #5702
